ecmascript/transforms/react/src/pure_annotations/tests.rs

```diff
@@ -0,0 +1,309 @@
+use super::*;
+use swc_common::comments::SingleThreadedComments;
+use swc_common::{sync::Lrc, FileName, SourceMap};
+use swc_ecma_codegen::{text_writer::JsWriter, Emitter};
+use swc_ecma_parser::{Parser, StringInput};
+use swc_ecma_transforms_base::resolver;
+use swc_ecma_transforms_testing::Tester;
+
+fn parse(
+    tester: &mut Tester,
+    src: &str,
+) -> Result<(Module, Lrc<SourceMap>, Lrc<SingleThreadedComments>), ()> {
+    let syntax = ::swc_ecma_parser::Syntax::Es(::swc_ecma_parser::EsConfig {
+        jsx: true,
+        dynamic_import: true,
+        ..Default::default()
+    });
+    let source_map = Lrc::new(SourceMap::default());
+    let source_file = source_map.new_source_file(FileName::Anon, src.into());
+
+    let comments = Lrc::new(SingleThreadedComments::default());
+    let module = {
+        let mut p = Parser::new(syntax, StringInput::from(&*source_file), Some(&comments));
+        let res = p
+            .parse_module()
+            .map_err(|e| e.into_diagnostic(&tester.handler).emit());
+
+        for e in p.take_errors() {
+            e.into_diagnostic(&tester.handler).emit()
+        }
+
+        res?
+    };
+
+    Ok((module, source_map, comments))
+}
+
+fn emit(
+    source_map: Lrc<SourceMap>,
+    comments: Lrc<SingleThreadedComments>,
+    program: &Module,
+) -> String {
+    let mut src_map_buf = vec![];
+    let mut buf = vec![];
+    {
+        let writer = Box::new(JsWriter::new(
+            source_map.clone(),
+            "\n",
+            &mut buf,
+            Some(&mut src_map_buf),
+        ));
+        let mut emitter = Emitter {
+            cfg: Default::default(),
+            comments: Some(&comments),
+            cm: source_map.clone(),
+            wr: writer,
+        };
+        emitter.emit_module(&program).unwrap();
+    }
+
+    return String::from_utf8(buf).unwrap();
+}
+
+fn run_test(input: &str, expected: &str) {
```

Comment:
I didn't find a way to get the default Tester to pass through comments correctly, so ended up duplicating some of the logic here. Is there a better way?

Reply:
Adding a field to `Tester` might work.

Reply:
Yeah, I tried that. There ended up being too many places to change. e.g. Tester only has one comments and source map instance, but really the actual and expected code need separate instances. Plus the `DropSpan` pass that's included in the tester breaks comments because it removes the spans that they are attached to. It just got too complicated with too many changes IMO so I bailed. Perhaps some of this logic I copied could be made reusable though. Just wondered if you had ideas or other places I didn't know about that did something similar.

Reply:
I see.

I used integration test for such tests, but it takes long time to compile.

DropSpan issue can be workarounded by using `eq_ignore_span`, which is added much later than `DropSpan`, but it may break lots of tests.

So your approach seems reasonable to me.

Thanks!